.github/workflows/update.yml

```
@@ -34,6 +34,7 @@ jobs:
     name: Update the flake inputs and generate options
     runs-on: ubuntu-latest
     timeout-minutes: 40
+    if: github.repository == 'nix-community/nixvim'
```

This would still allow users to run manually, using workflow despatch. IDK _why_ they'd want to do that, maybe just to test things out?

```suggestion
    if: github.event_name != 'schedule' || github.repository == 'nix-community/nixvim'
```

i.e. `event_name == 'schedule' -> repo == 'nix-community/nixvim'`

---

It shouldn't allow a non maintainer to workflow dispatch, right? If that was the concern. Otherwise, it'd be nice to have flexibility to run ad-hoc when there is a fix from upstream for stuff broken preventing us from updating on the current nixpkgs revision.

---

Those kinda security concerns are handled by GitHub's permission system. We don't need to deal with them in the workflow; in fact we can't, since anyone is free to fork and change the workflow content.

Rather, we just want to prevent _automatically_ running the workflow on forks.

Any workflow runs on a fork will only affect the fork repo, but it is annoying to have it run automatically.

---

Yeah, my question was that I wasn't sure of the motivation for the requested change. This prevents us from running it on demand in this repo.

---

> This would still allow users to run manually, using workflow despatch. IDK _why_ they'd want to do that, maybe just to test things out?
> 
> i.e. `event_name == 'schedule' -> repo == 'nix-community/nixvim'`

Sorry, I think I totally misread this... I see you were saying you wanted them to be able to run on demand on their own repo with the change and it's an OR condition so it will still work for us.